File: lib/DevOther/src/main/java/dev/engine/cache/DataItem.java
package dev.engine.cache;

/**
 * detail: Cache Data Item
 * @author Ttt
 */
public class DataItem
        extends ICacheEngine.EngineItem {

    public DataItem(
            String key,
            int type,
            long size,
            long saveTime,
            long validTime,
            boolean isPermanent,
            boolean isDue
    ) {
        super(key, type, size, saveTime, validTime, isPermanent, isDue);
    }
}

File: lib/DevOther/src/main/java/ktx/dev/engine/cache/DataItem.kt
package ktx.dev.engine.cache

import dev.engine.cache.ICacheEngine

/**
 * detail: Cache Data Item
 * @author Ttt
 */
class DataItem(
    key: String?,
    type: Int,
    size: Long,
    saveTime: Long,
    validTime: Long,
    isPermanent: Boolean,
    isDue: Boolean
) : ICacheEngine.EngineItem(key, type, size, saveTime, validTime, isPermanent, isDue)